sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/viewstate/messageholder/HolderBindingExtensions.kt

```diff
@@ -813,77 +812,89 @@ internal inline fun LayoutMessageHolderBinding.setBubbleBotResponse(
                 }
-                MessageType.Purchase.Denied -> {
-                    R.string.purchase_status_label_paid_message_details_denied
+                PurchaseStatus.Accepted -> {
+                    getString(R.string.purchase_status_label_paid_message_details_accepted)
                 }
-                MessageType.Purchase.Processing -> {
-                    R.string.purchase_status_label_paid_message_details_processing
+                PurchaseStatus.Denied -> {
+                    getString(R.string.purchase_status_label_paid_message_details_denied)
                 }
-                null -> {
-                    R.string.purchase_status_label_paid_message_details_default
+                else -> {
+                    getString(R.string.purchase_status_label_paid_message_details_default)
+                }
+            }
+
+            val statusIcon: String = when (paidDetails.purchaseStatus) {
+                PurchaseStatus.Accepted -> {
+                    getString(R.string.material_icon_name_payment_accepted)
+                }
+                PurchaseStatus.Denied -> {
+                    getString(R.string.material_icon_name_payment_denied)
+                }
+                else -> {
+                    ""
                 }
             }
 
             backgroundDrawableResId?.let { root.setBackgroundResource(it) }
             root.backgroundTintList = ContextCompat.getColorStateList(root.context, backgroundTintResId)
 
-            imageViewPaidMessageReceivedIcon.goneIfFalse(paidDetails.showPaymentReceivedIcon)
-            imageViewPaidMessageSentIcon.goneIfFalse(paidDetails.showSendPaymentIcon)
-            textViewPaymentAcceptedIcon.goneIfFalse(paidDetails.showPaymentAcceptedIcon)
-            progressBarPaidMessage.goneIfFalse(paidDetails.showPaymentProgressWheel)
-            textViewPaidMessageStatusLabel.text = getString(statusTextResID)
+            textViewPaymentStatusIcon.text = statusIcon
+            textViewPaymentStatusIcon.goneIfFalse(
+                paidDetails.purchaseStatus.isPurchaseAccepted() ||
+                        paidDetails.purchaseStatus.isPurchaseDenied()
+            )
+
+            progressBarPaidMessage.goneIfFalse(paidDetails.purchaseStatus.isPurchaseProcessing())
+
+            textViewPaidMessageStatusLabel.goneIfFalse(
+                paidDetails.purchaseStatus.isPurchaseProcessing() ||
+                        paidDetails.purchaseStatus.isPurchaseAccepted() ||
+                        paidDetails.purchaseStatus.isPurchaseDenied()
+            )
+            textViewPaidMessageStatusLabel.text = statusText
+
+            buttonPayAttachment.goneIfFalse(paidDetails.purchaseStatus.isPurchasePending())
+            textViewPayMessageLabel.goneIfFalse(paidDetails.purchaseStatus.isPurchasePending())
+            imageViewPayMessageIcon.goneIfFalse(paidDetails.purchaseStatus.isPurchasePending())
```

Comment:
Move all layout visibility logic to `LayoutState.Bubble.ContainerFourth.PaidMessageReceivedDetails` object that will get set upon instantiation of the `MessageHolderViewState`. Some variables on that LayoutState like, `val showStatusIcon: Boolean, val showButtonPayAttachment: Boolean, ....`